### PR TITLE
Markets data tray

### PR DESCRIPTION
**Pull request by pixelandpage**
Makes the markets data tray from acceleration month persist in an open/closed state.
Updated flag = homepageMarketsDataTray
 
Design tweaks: 
 - increase the clickable link areas
 - improve tabs spacing
 - update font sizes 
 - update breakpoints to prevent overlapping text

### CODE REVIEW

File: client/components/markets-data-details/components/component.js [hunk 1]
```diff
@@ -49,14 +48,20 @@ export default class Component {
...
 				? this.element.className = this.element.className.replace('mdd-hide', '')
 				: this.element.className += ' mdd-hide';
 		}
 	}
+
+		toggleQuotesStripVisibility (component) {
+			if (component.tabs.element.className) {
+				component.tabs.element.className.includes('mdd-hide') ?
+				component.quotesStrip.element.className = component.quotesStrip.element.className.replace('mdd-hide', '') :
+				component.quotesStrip.element.className += ' mdd-hide';
```
Comment (leggsimon) [inline client/components/markets-data-details/components/component.js]: Could use `component.quotesStrip.element.classlist.add('mdd-hide');` here

File: client/components/markets-data-details/components/component.js [hunk 2]
```diff
@@ -49,14 +48,20 @@ export default class Component {
 	}
 
 	toggleVisibility () {
-
 		if(this.element) {
-
 			this.element.className.includes('mdd-hide')
 				? this.element.className = this.element.className.replace('mdd-hide', '')
```
Comment (leggsimon) [inline client/components/markets-data-details/components/component.js]: `this.element.classlist.remove('mdd-hide')`

File: client/components/markets-data-details/components/component.js [hunk 3]
```diff
@@ -49,14 +48,20 @@ export default class Component {
...
 				: this.element.className += ' mdd-hide';
```
Comment (leggsimon) [inline client/components/markets-data-details/components/component.js]: `this.element.classlist.add('mdd-hide')`

File: client/components/markets-data-details/components/details/details.js [hunk 4]
```diff
@@ -12,30 +14,29 @@ export default class MarketsDataDetails extends Component {
 
 		this.tabs = new Tabs({
 			id: 'mddTabs',
-			isVisible: false
+			isVisible: showMddTray() || false
 		});
 
 		this.quotesStrip = new QuotesStrip({
-			id: 'mddQuotesStrip'
+			id: 'mddQuotesStrip',
```
Comment (leggsimon) [inline client/components/markets-data-details/components/details/details.js]: I don't think we use the trailing commas convention so should probably remove this

File: client/components/markets-data-details/components/details/details.scss [hunk 5]
```diff
@@ -64,15 +64,15 @@
 }
 
 
-@include oGridRespondTo(S) {
+@include oGridRespondTo(M) {
 		.markets-data-details__minute-delay-text {
 				display: inline;
 		}
 }
```
Comment (leggsimon) [inline client/components/markets-data-details/components/details/details.scss]: This block has double tabbed indentation... I know it's not your changes but while you're there..

File: client/components/markets-data-details/components/details/details.js [hunk 6]
```diff
@@ -68,7 +69,7 @@ export default class MarketsDataDetails extends Component {
 
 function assembleTabsFromMarkets (markets) {
 
-	return markets.map(market => {
+return markets.map(market => {
 		const tab = new QuoteTabViewModel();
```
Comment (leggsimon) [inline client/components/markets-data-details/components/details/details.js]: double tabbed indentation?

File: client/components/markets-data-details/components/details/details.scss [hunk 7]
```diff
@@ -38,7 +38,7 @@
...
 
 .markets-data-details__footer-text {
-		font-size: 14px;
+		font-size: 12px;
```
Comment (leggsimon) [inline client/components/markets-data-details/components/details/details.scss]: double tabbed indentation?

File: client/components/markets-data-details/components/quote/quote.scss [hunk 8]
```diff
@@ -11,91 +11,34 @@
...
-		&:visited {
-				color: white;
-		}
-}
 
-.mdd-quote__price-link,
 .mdd-quote__price {
 		color: #b5b5b5;
-		font-size: 16px;
+		font-size: 12px;
```
Comment (leggsimon) [inline client/components/markets-data-details/components/quote/quote.scss]: double tabbed indentation?

File: client/components/markets-data-details/components/quote/quote.scss [hunk 9]
```diff
@@ -11,91 +11,34 @@
...
-		&:visited {
-				color: #b5b5b5;
-		}
 }
 
-.mdd-quote__percentage-change-link,
 .mdd-quote__percentage-change {
 		color: #9cd321;
-		font-size: 16px;
+		font-size: 12px;
```
Comment (leggsimon) [inline client/components/markets-data-details/components/quote/quote.scss]: double tabbed indentation?

File: client/components/markets-data-details/components/quote/quote.scss [hunk 10]
```diff
@@ -11,91 +11,34 @@
...
-		}
-
-		&:visited {
-				color: #9cd321;
-		}
 }
 
 .mdd-quote__percentage-change--negative {
 		color: #ff7287;
 		border-bottom-width: 0;
```
Comment (leggsimon) [inline client/components/markets-data-details/components/quote/quote.scss]: double tabbed indentation? Gawd! It's everywhere!

File: client/components/markets-data-details/components/details/details.js (hunk 4 above)
Comment (debugwand) [inline client/components/markets-data-details/components/details/details.js]: stray comma

File: client/components/markets-data-details/components/toggle-btn/toggle-btn.js [hunk 11]
```diff
@@ -1,44 +1,58 @@
...
+	: element.innerHTML = EXPAND_WORDING;
 }
 
 function toggleIconRotation (component) {
 	const element = component.element.querySelector('.mdd-toggle-btn__icon');
 
 	element.className.includes('mdd-rotate-180')
-		? element.className = element.className.replace('mdd-rotate-180', '')
-		: element.className += ' mdd-rotate-180';
+	? element.className = element.className.replace('mdd-rotate-180', '')
```
Comment (leggsimon) [inline client/components/markets-data-details/components/toggle-btn/toggle-btn.js]: `element.classlist.remove('mdd-rotate-180')`

File: client/components/markets-data-details/components/toggle-btn/toggle-btn.js [hunk 12]
```diff
@@ -1,44 +1,58 @@
...
+	: element.className += ' mdd-rotate-180';
```
Comment (leggsimon) [inline client/components/markets-data-details/components/toggle-btn/toggle-btn.js]: `element.classlist.add('mdd-rotate-180')`

File: client/components/markets-data-details/components/component.js [hunk 13]
```diff
@@ -49,14 +48,20 @@ export default class Component {
...
 		}
 	}
+
+		toggleQuotesStripVisibility (component) {
```
Comment (debugwand) [inline client/components/markets-data-details/components/component.js]: keep with the code that 'component' refers to

File: client/components/markets-data-details/components/quote/quote.scss [hunk 14]
```diff
@@ -11,91 +11,34 @@
...
-				color: white;
-		}
-
-		&:visited {
-				color: white;
-		}
-}
 
-.mdd-quote__price-link,
 .mdd-quote__price {
```
Comment (debugwand) [inline client/components/markets-data-details/components/quote/quote.scss]: links need indication for keyboard users

File: client/components/markets-data-details/components/component.js (hunk 2 above)
Comment (debugwand) [inline client/components/markets-data-details/components/component.js]: classlist not supported in IE9

Comment (leggsimon) [inline client/components/markets-data-details/components/component.js]: Ahh, thanks for checking 🤦‍♂️ I didn't think of that!

**Comment by pixelandpage** [comment]
Good spot re: not using origami colours - I double checked this with design, it's deliberate pending the upcoming big rethink of o-colors 👍

**Pull request merged**